atomisticparsers/h5md/parser.py

```diff
@@ -0,0 +1,787 @@
+                    property_value_key = property_type_dict['properties_known'][observable_type]
+                    flag_known_property = True
+                workflow_results[property_key] = []
+                property_dict = {property_value_key: []}
+                property_dict['label'] = observable_type
+                for key, observable in observable_dict.items():
+                    property_values_dict = {'label': key}
+                    for quant_name, val in observable.items():
+                        if quant_name == 'val':
+                            continue
+                        if quant_name == 'bins':
+                            continue
+                        if quant_name in property_type_dict['property_keys_list']:
+                            property_dict[quant_name] = val
+                        if quant_name in property_type_dict['property_value_keys_list']:
+                            property_values_dict[quant_name] = val
+                        # TODO Still need to add custom values here.
+
+                    val = observable.get('value')
+                    if val is not None:
+
+                        value_unit = val.units if hasattr(val, 'units') else None
+                        value_magnitude = val.magnitude if hasattr(val, 'units') else val
+                        if flag_known_property:
+                            property_values_dict['value'] = value_magnitude * value_unit if value_unit else value_magnitude
+                        else:
+                            property_values_dict['value_unit'] = str(value_unit) if value_unit else None
+                            property_values_dict['value_magnitude'] = value_magnitude
+
+                    bins = observable.get('bins')
+                    if bins is not None:
+                        bins_unit = bins.units if hasattr(bins, 'units') else None
+                        bins_magnitude = bins.magnitude if hasattr(bins, 'units') else bins
+                        if flag_known_property:
+                            property_values_dict['bins'] = bins_magnitude * bins_unit if bins_unit else bins_magnitude
+                        else:
+                            property_values_dict['bins_unit'] = str(bins_unit) if bins_unit else None
+                            property_values_dict['bins_magnitude'] = bins_magnitude
+
+                    property_dict[property_value_key].append(property_values_dict)
+                workflow_results[property_key].append(property_dict)
+
+        workflow_results = {}
+        ensemble_average_observables = self.observable_info.get('ensemble_average')
+        ensemble_property_dict = {
+            'property_type_key': 'ensemble_properties',
+            'property_type_value_key': 'ensemble_property_values',
+            'properties_known': {'radial_distribution_functions': 'radial_distribution_function_values'},
+            'property_keys_list': EnsembleProperty.__dict__.keys(),
```

Comment:
use EnsembleProperty.m_def.all_quantities.keys(), same for Correlation

Reply:
thanks. done.